src/models/users.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+class Users {
+  constructor() {
+    this.users = [];
+  }
+
+  reset() {
+    this.users = [];
+  }
+
+  getUsers() {
+    return this.users;
+  }
+
+  getUser(id) {
+    return this.users.find((user) => user.id === id);
+  }
+
+  addUser({ name }) {
+    const newUser = {
+      id: this.users.length + 1,
+      name,
+    };
+
+    this.users.push(newUser);
+
+    return newUser;
+  }
+
+  removeUser(id) {
+    const user = this.getUser(id);
+
+    if (!user) {
+      return null;
+    }
+
+    this.users = this.users.filter((currentUser) => currentUser.id !== id);
+
+    return user;
+  }
+
+  updateUser({ id, name }) {
+    const user = this.users.find((currentUser) => currentUser.id === +id);
```

Comment:
The `id` parameter is converted to a number using `+id`, but this conversion is not consistently applied in other methods like `getUser` and `removeUser`. Consider ensuring consistent type handling for `id` across all methods.